client/my-sites/domains/domain-management/transfer/transfer-to-other-site/transfer-domain-to-other-site.tsx

```diff
@@ -0,0 +1,236 @@
+	TransferDomainToOtherSitePassedProps,
+	TransferDomainToOtherSiteProps,
+	TransferDomainToOtherSiteStateProps,
+} from './types';
+
+import './style.scss';
+
+const wpcom = wp.undocumented();
+
+export class TransferDomainToOtherSite extends Component< TransferDomainToOtherSiteProps > {
+	state = {
+		targetSiteId: null,
+		showConfirmationDialog: false,
+		disableDialogButtons: false,
+	};
+
+	isDataReady(): boolean {
+		return ! this.props.isRequestingSiteDomains && this.props.hasSiteDomainsLoaded;
+	}
+
+	isSiteEligible = ( site: TransferDomainToOtherSiteProps[ 'selectedSite' ] ): boolean => {
+		// check if it's an Atomic site from the site options
+		const isAtomic = site?.options?.is_automated_transfer ?? false;
+
+		return (
+			site.capabilities.manage_options &&
+			! ( site.jetpack && ! isAtomic ) && // Simple and Atomic sites. Not Jetpack sites.
+			! ( site?.options?.is_domain_only ?? false ) &&
+			site.ID !== this.props.selectedSite.ID
+		);
+	};
+
+	handleSiteSelect = ( targetSiteId: number ): void => {
+		this.setState( {
+			targetSiteId,
+			showConfirmationDialog: true,
+		} );
+	};
+
+	handleConfirmTransfer = (
+		targetSite: TransferDomainToOtherSiteProps[ 'selectedSite' ],
+		closeDialog: () => void
+	): void => {
+		const { selectedDomainName } = this.props;
+		const targetSiteTitle = targetSite.title;
+		const successMessage = this.props.translate(
+			'%(selectedDomainName)s has been transferred to site: %(targetSiteTitle)s',
+			{ args: { selectedDomainName, targetSiteTitle } }
+		);
+		const defaultErrorMessage = this.props.translate(
+			'Failed to transfer %(selectedDomainName)s, please try again or contact support.',
+			{
+				args: { selectedDomainName },
+			}
+		);
+
+		this.setState( { disableDialogButtons: true } );
+		wpcom
+			.transferToSite( this.props.selectedSite.ID, this.props.selectedDomainName, targetSite.ID )
+			.then(
+				() => {
+					this.props.successNotice( successMessage, { duration: 10000, isPersistent: true } );
+					if ( this.props.isDomainOnly ) {
+						this.props.requestSites();
+						const transferedTo = this.props.sites!.find( ( site ) => site!.ID === targetSite.ID );
+						page( domainManagementList( transferedTo!.slug ) );
+					} else {
+						page( domainManagementList( this.props.selectedSite.slug ) );
+					}
+				},
+				( error: Error ) => {
+					this.setState( { disableDialogButtons: false } );
+					closeDialog();
+					this.props.errorNotice( error.message || defaultErrorMessage );
+				}
+			);
+	};
+
+	handleDialogClose = (): void => {
+		if ( ! this.state.disableDialogButtons ) {
+			this.setState( { showConfirmationDialog: false } );
+		}
+	};
+
+	getMessage(): i18nCalypso.TranslateResult {
+		const { selectedDomainName: domainName, translate } = this.props;
+		const translateArgs = { args: { domainName }, components: { strong: <strong /> } };
+		return translate(
+			"Transfer {{strong}}%(domainName)s{{/strong}} to a site you're an administrator of:",
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **22** times:
`translate( 'Please choose a site you're an administrator on to transfer {{strong}}%(domainName)s{{/strong}} to:' )` *ES Score: 6*